vector/src/main/java/im/vector/app/features/spaces/explore/SpaceDirectoryController.kt

```diff
@@ -209,4 +211,7 @@ class SpaceDirectoryController @Inject constructor(
             }
         }
     }
+
+    private fun SpaceChildInfo.isUpgradedRoom(data: SpaceDirectoryState) =
+            data.knownRoomSummaries.any { it.roomId == childRoomId && it.versioningState != VersioningState.NONE }
```

Comment:
super nit: could be a little easier on the brain to lift the `isUpgraded = it.versioningState != VersioningState.NONE` to an extension on the `VersioningState` 

has the benefit of keeping the conditions here positive and future proofs this logic from any possible new _versioning states_ which aren't upgrades

Reply:
I agree, good shout!